cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/CollectMacroEvaluator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.plugin;
+
+import co.cask.cdap.api.macro.InvalidMacroException;
+import co.cask.cdap.api.macro.MacroEvaluator;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Macro evaluator that simply collects the macro properties in a set
+ */
+public class CollectMacroEvaluator implements MacroEvaluator {
+  private final Set<String> propertiesMacros;
+
+  @VisibleForTesting
+  CollectMacroEvaluator() {
+    propertiesMacros = new HashSet<>();
+  }
+
+  @VisibleForTesting
+  Set<String> getPropertiesMacros() {
```

Comment:
nit: I think its clearer to call these lookup macros, since thats the method name in the API.